Source/Engine/Core/Collections/Array.h

```diff
@@ -323,7 +312,7 @@ API_CLASS(InBuild) class Array
     FORCE_INLINE const T& operator[](int32 index) const
     {
         ASSERT(index >= 0 && index < _count);
-        return _allocation.Get()[index];
+        return Get()[index];
```

Comment:
I'm aware that changing `_allocation.Get()` into `Get()` makes code cleaner and more readable but it comes at cost of performance - esp. in Debug builds where MSVC compiler doesn't inline that call.

Reply:
Actually, I'm going to reverse a lot of those changes. I guess including this one. I got a little bit of tunnel vision trying to get rid of the type template argument from allocation data. On the positive note, now I understand the internals of the collections significantly better. I'm definitely splitting the PR into more digestible portions both for me, cr'ers and the users.